io.c

```diff
@@ -2349,15 +2339,15 @@ struct file_operations io_ring_fops = {
 	.mmap = io_uring_mmap,
 };
 
-int io_ring_register_device(char *name, struct io_ring_ctx *ctx, int context_id)
+int io_ring_register_device(struct io_dev *ctx, int context_id)
 {
 	struct miscdevice *dev = &ctx->miscdev;
 
 	ctx->context_id = context_id;
 
-	sprintf(name, "pxd/pxd-io-%d", context_id);
+	sprintf(ctx->name, "pxd/pxd-io-%d", context_id);
```

Comment:
ctx->name is defined 256 bytes, but this name format will not go more than 22 bytes approx. Maybe can be reduced to 64 chars atleast... even with sprintf usage it should suffice isn't it?